File: app/src/main/java/com/mglabs/eventme/EventAdapter.java
package com.mglabs.eventme;

import android.content.Context;
import android.content.Intent;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.bumptech.glide.Glide;
import com.mglabs.eventme.model.Event;
import com.mglabs.eventme.model.EventStore;

import org.w3c.dom.Text;

import java.util.ArrayList;
import java.util.List;

public class EventAdapter extends RecyclerView.Adapter<EventAdapter.EventViewHolder> {

    //Member variables
    EventStore store;
    private LayoutInflater mInflater;
    Event mCurrentEvent;
    private Context mContext;

    public static String EVENT_ID_EXTRA = " ";
    public static String EVENT_IMAGE_EXTRA = "";

    //Adapter constructor
    public EventAdapter(Context context, EventStore store) {
        mInflater = LayoutInflater.from(context);
        this.store = store;
        this.mContext = context;
    }

    @NonNull
    @Override
    public EventViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = mInflater.inflate(R.layout.list_item, parent, false);
        return new EventViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull EventViewHolder eventViewHolder, int position) {
        if (store.getEventList() != null) {
            Event mCurrentEvent = store.getEventList().get(position);
            eventViewHolder.bindTo(mCurrentEvent);
        }
    }

    @Override
    public int getItemCount() {
        if (store.getEventList() != null)
            return store.getEventList().size();
        else
            return 0;
    }

    /**
     * ViewHolder class that represents each row of data in the RecyclerView
     */
    public class EventViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        final TextView eventTitle;
        final ImageView eventImage;

        /**
         * Constructor for the ViewHolder, used in onCreateViewHolder().
         *
         * @param itemView The rootview of the list_item.xml layout file
         */
        public EventViewHolder(@NonNull View itemView) {
            super(itemView);

            //Initialize the views
            eventTitle = itemView.findViewById(R.id.title_textView);
            eventImage = itemView.findViewById(R.id.event_image);

            //Set the onClickListener on the entire view
            itemView.setOnClickListener(this);
        }

        void bindTo(Event currentEvent) {
            //Populate the textviews with data and images with Glide
            eventTitle.setText(currentEvent.getTitle());
            //eventInfo.setText(currentEvent.getInfo());
            Glide.with(mContext).load(currentEvent.getImageResource()).into(eventImage);
        }

        @Override
        public void onClick(View view) {
            if (store.getEventList() != null) {
                int position = getAdapterPosition();
                Event event = store.getEventList().get(position);
                Intent intent = new Intent(mContext, DetailActivity.class);
                intent.putExtra(EVENT_ID_EXTRA, event.getEventID());
                intent.putExtra(EVENT_IMAGE_EXTRA, event.getImageResource());
                mContext.startActivity(intent);

                Log.i("ID_INFO", "onClick:" + event.getInfo());

            } else {
                Toast.makeText(mContext, "EVENT NOT FOUND", Toast.LENGTH_SHORT).show();
            }
        }
    }
}

File: app/src/main/java/com/mglabs/eventme/ui/events/EventAdapter.kt
package com.mglabs.eventme.ui.events

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.mglabs.eventme.R
import com.mglabs.eventme.data.model.Event
import com.mglabs.eventme.databinding.EventListItemBinding

class EventAdapter(private val eventList: List<Event>, private val listener: ListItemListener) :
    RecyclerView.Adapter<EventAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): EventAdapter.ViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val view = inflater.inflate(R.layout.event_list_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: EventAdapter.ViewHolder, position: Int) {
        val event = eventList[position]

       with(holder.binding) {
            eventTitleTv.text = event.title
           // eventLocationTv.text = event.
            root.setOnClickListener {
                // I'm notifying the listener class (the fragment) that the user has clicked on that element
                listener.editEvent(event.eventId?: "Evento non trovato")
            }
       }
    }

    override fun getItemCount(): Int {
        return eventList.size
    }


    // ViewHolder
    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        // Voglio riferirmi al layout della row list_item.xml
        val binding = EventListItemBinding.bind(itemView)
    }

    //Get the click event on the edit icon
    interface ListItemListener {
        fun editEvent(eventId: String)
    }
}